Deferred.js

```diff
@@ -262,6 +262,63 @@ define([
 			return listener.deferred.promise;
 		};
 
+		this["finally"] = promise["finally"] = function(callback) {
+			// summary:
+			//		Add a callback to the deferred that will fire whether it
+			//		resolves or rejects.
+			// description:
+			//		Conforms to ES2018's `Promise.prototype.finally`.
+			//		Add a callback to the deferred that will fire whether it
+			//		resolves or rejects. No value is passed to the callback.
+			//		Returns a promise that reflects the state of the original deferred,
+			//		with two exceptions:
+			//		- If the callback return a promise, the outer promise will wait
+			//		until the returned promise is resolved, then it will resolve
+			//		with the original value.
+			//		- If the callback throws an exception or returns a promise that
+			//		is rejected (or rejects later), the outer promise will reject
+			//		with the inner promise's rejection reason.
+			// callback: Function?
+			//		Callback to be invoked when the deferred is resolved
+			//		or rejected. Doesn't receive any value.
+			// returns: dojo/promise/Promise
+			//		Returns a new promise that reflects the state of the original deferred,
+			//		with two small exceptions (see description).
+			//
+
+			var listener = [
+				null,
+				function (value){
+					var valueOrPromise = callback();
+					if (valueOrPromise && typeof valueOrPromise.then === "function"){
+						return valueOrPromise.then(function (){
+							return value;
+						});
+					}
+					return value;
+				},
+				function (reason){
+					var valueOrPromise = callback();
+					if (valueOrPromise && typeof valueOrPromise.then === "function"){
+						return valueOrPromise.then(function (){
+							return new Deferred().reject(reason);
```

Comment:
@dylans @wkeese An alternative here (also on line 307) is to use `throw reason`. There may be a slight performance difference, but that may vary between platforms.

Any preference?

Reply:
I don't know enough about the code to comment...

Reply:
@dasa Anyone from your team want to take a look?

Reply:
I'd probably go with `throw reason;` since I don't see any other code in this module using `new Deferred().reject` but there is code that throws in strict mode.

Reply:
Thank @dasa. I switched to `throw reason`